test/models/user_test.rb

```diff
@@ -95,4 +95,15 @@ def setup
     @user.reload
     assert_equal 'supervolunteer@example.com', @user.email
   end
+
+  test 'updating client mail doesnt change current user email' do
+    @superadmin = create :user, role: 'superadmin', email: 'superadmin@example.com'
```

Comment:
There is already `@superadmin` at the setup. You could have only added her email here. (Although by default the `create :user` creates a superadmin, I don't know why we need two.)